### ISSUE TITLE
SASutil.hpimpute doesn't recognize the "impute" parameter.

### ISSUE THREAD

**Issue by jlwalke2** [sassoftware/saspy#14]
```
sas.teach_me_SAS(True)
util.hpimpute(data=cars, input='input', impute='impute', performance='performance', id='id', freq='freq', code='code')
```

> %macro proccall(d);
proc hpimpute data=WORK.cars   ;
code file='code';
freq freq;
id id;
input input;
performance performance;
run; quit; %mend;
%mangobj(hpi0011,hpimpute,cars);

**Comment by FriedEgg** [comment]
Confirmed.  Looks like 'impute' is missing from _makeProcCallMacro in the SASProcCommons class

**Comment by jld23** [comment]
In thinking about this. It makes more sense to me to add an impute (and binning) method to the SASData object since that is where imputation happens.
Here is my proposed signature:
```
    def impute(self, vars: dict, replace: bool = False, prefix: str = 'imp_', out: 'SASData' = None) -> 'SASdata':
        """
        Imputes missing values for a SASdata object. 

        :param vars: a dictionary in the form of {'varname':'impute type'} or {'impute type':'[var1, var2]'}
        :param replace:
        :param prefix:
        :param out:
        :return:
        """
```

I'm not sure what format makes the most sense for identifying the vars and type of imputation to be performed. Here are the best I've come up with:
`{'varname':'impute type'} or {'impute type':'[var1, var2]'}`

**Comment by FriedEgg** [comment]
Personally, I feel that the later idea is better `{'impute type':'[var1, var2]'}` since it allows for the list of variables per imputation method.  

My only question would be, how would you define a VALUE imputation instead of a METHOD imputation?

```
proc hpimpute data=sampsio.hmeq out=out1;
     input mortdue value clage debtinc;
     impute mortdue / value  = 70000;
     impute value   / method = mean;
     impute clage   / method = random;
     impute debtinc / method = pmedian;
run;
```

**Comment by jld23** [comment]
I think this is why this hasn't been done yet --  because I agree the first is better but doesn't allow for imputation of constants.

Let's think on it a bit more.

**Comment by jld23** [comment]
I think I have a good compromise between the issues.
The dictionary will have the imputation type as the key and a list of variables and for `value` it will be a list of tuples (variable name, constant)

Here is an example:

```
vars = {'mean'   : ['value'],
        'random' : ['clage'],
        'pmedian': ['debtinc'],
        'value'  : [('mortdue',70000), ('job', 'other')]}
```

**Comment by FriedEgg** [comment]
That seems totally reasonable, I'm not just completely clear what the second tuple in the value imputation example you gave would mean?  My assumption is that it would create a statement

```impute job / value='other';```

and not

``impute mortdue job other / value=70000;```

The impute statement only excepts numeric variables, which is the source of my issue.  I would assume to get the second statement I provided above the correct syntax would be:

```vars = {'value': [('mortdue','job','other',70000)]}```

and 

```vars = {'value': [('mortdue',70000), ('job', 1)]}```

would also be valid

**Comment by jld23** [comment]
@FriedEgg @jlwalke2 
Do you want to take a look at my PR and see if you have any questions?

Here is some sample code using HMEQ

```
impute_dict = {'mean'     : ['value'],
               'random'   : ['clage'],
               'median'   : ['debtinc'],
               'midrange' : ['clno', 'yoj'],
               'max'      : ['ninq'],
               'min'      : ['delinq'],
               'mode'     : ['Job'],
               'value'    : [('mortdue',70000), ('loan', 9999)]}
outds = sas.sasdata('outds')

hmeq_test.impute(vars=impute_dict, replace=True, out=outds)
hmeq_test.impute(vars=impute_dict, replace=True, out="foo")
hmeq_test.impute(vars=impute_dict, replace=True, out="foo.bar")
hmeq_test.impute(vars=impute_dict, replace=True)
hmeq_test.impute(vars=impute_dict, replace=True, prefix='_t_')
hmeq_test.impute(vars=impute_dict)
```

**Comment by jld23** [comment]
Sorry I forgot the hpimpute code

```
util = sas.sasutil()
impute_dict = {'mean'   : ['value'],
        'random' : ['clage'],
        'pmedian': ['debtinc'],
        'value'  : [('mortdue',70000)]}

imp = util.hpimpute(data=hmeq, impute = impute_dict )
```
This is the SAS code output:
```
proc hpimpute data=d.hmeq   ;

input value debtinc clage;
impute value / method = mean;
impute clage / method = random;
impute debtinc / method = pmedian;
run;run; quit;
```

**Comment by FriedEgg** [comment]
@jld23 Judging solely by what you posted above, I do not see the value imputation for variable mortdue as defined in the impute_dict represented in the SAS code

**Comment by jld23** [comment]
@FriedEgg chaning missing mortude to 70,000 doesn't make scense I agree, I just wanted to show how a user would format a constant imputation for a variable since it uses a list of tuples not just a list.

**Comment by jld23** [comment]
This is fixed by PR #32
